src/server/object_services/schemas/object_part_indexes.js

```diff
@@ -31,4 +32,18 @@ module.exports = [{
             }
         }
     },
+    {
+        // This index is used for queries where we want to find all the chunks of a specific object
+        // which are already marked deleted.
+        fields: {
+            obj: 1,
```

Comment:
@utkarsh-pro IIUC, this can also be achieved by removing the filter on `deleted`, Is that right? I wonder which approach is better, since adding indexes can affect writes performance. @guymguym any thoughts?

Reply:
@dannyzaken, I think partial indexes are faster for both reads and writes.

> One major reason for using a partial index is to avoid indexing common values. Since a query searching for a common value (one that accounts for more than a few percent of all the table rows) will not use the index anyway, there is no point in keeping those rows in the index at all. This reduces the size of the index, which will speed up those queries that do use the index. It will also speed up many table update operations because the index does not need to be updated in all cases. [Example 11.1](https://www.postgresql.org/docs/current/indexes-partial.html#INDEXES-PARTIAL-EX1) shows a possible application of this idea.

Taken from: https://www.postgresql.org/docs/current/indexes-partial.html

Reply:
Or did I... misunderstood the comment?

Reply:
We discussed these new indexes on parts/multiparts and while there is an alternative to define the index only on the `deleted` field like in objectmds aggregate_by_delete_dates, this would mean that db_cleaner will cleanup parts and multiparts without specific relation to a deleted object, and this could be fine, but we don't have to do it now. The benefit is that the index will allow an index condition on the delete date, but the downside is that we will delete some parts and multiparts of objects and not all parts of the object, which might look a bit weird for other scanners (perhaps). In any case, this looks fine as we have to have some index for deletion, and this index will be limited by the amount of pending rows to deleted (the db_cleaner will keep it small).